github.go

```diff
@@ -92,38 +92,10 @@ func RunGithubHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, "OK")
 }
 
-func (base GithubImporter) GetProvider(i Import) (wharfapi.Provider, error) {
-	var provider wharfapi.Provider
-	var err error
-	if i.ProviderId != 0 {
-		provider, err = base.WharfClient.GetProviderById(i.ProviderId)
-		if err != nil {
-			return provider, err
-		} else if provider.ProviderID == 0 {
-			err = fmt.Errorf("provider with id %v not found", i.ProviderId)
-		} else if provider.URL != i.Url {
-			err = fmt.Errorf("invalid url in provider %v", provider.URL)
-		} else if provider.UploadURL != i.UploadUrl {
-			err = fmt.Errorf("invalid upload url in provider %v", provider.UploadURL)
-		}
-	} else {
-		provider, err = base.WharfClient.GetProvider("github", i.Url, i.UploadUrl, base.Token.TokenID)
-		if err != nil || provider.ProviderID == 0 {
-			provider, err = base.WharfClient.PostProvider(wharfapi.Provider{Name: "github", URL: i.Url, UploadURL: i.UploadUrl})
-		}
-	}
-	fmt.Println("Provider from db: ", provider)
-	return provider, nil
-}
-
 func (base GithubImporter) GetToken(i Import) (wharfapi.Token, error) {
 	var token wharfapi.Token
 	var err error
 
-	if base.Provider.ProviderID == 0 {
-		return token, fmt.Errorf("provider not found")
-	}
-
```

Comment:
There's still some code in this function relies on `base.Provider.ProviderID` but `base.Provider` is only set inside `GetProvider`.

You will have to take a second look at this function. Maybe rip out logic that doesn't make sense to keep, or revert back to fetching provider first and token second

Reply:
I pulled your branch and tested it and it works nicely. One thing though is that every time I import it recreates the token in the database, even if there was one identical it could reuse.

Sorry to bring this up again, but this is too bad of a flow to be accepted by me.

Reply:
Will start a PR on the wharf-api-client repo to create a PATCH call for the Token. Will link here when done.

Reply:
@jilleJr See the PR here https://github.com/iver-wharf/wharf-api-client-go/pull/8 

If it works like i think it does (i have not tested it yet working on that now) then perhaps a PUT instead of a POST could be the solution to part of the problem we spoke of here.

Reply:
Resolving this as its become stale. Links are up in the description.